### PR TITLE
Adding tests for CertsToRemove, and move tests around

### PR DESCRIPTION
**Pull request by diogomonica**
Adding tests for certs to remove and moving tests to the right file.

Signed-off-by: Diogo Monica <diogo@docker.com>

### CODE REVIEW

**Comment by diogomonica** [comment]
@NathanMcCauley @endophage

File: keystoremanager/keystoremanager_test.go [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+
+	signedTestRoot, err := testRoot.ToSigned()
+	assert.NoError(t, err)
+
+	// We only sign with the old key, and not with the new one
+	err = signed.Sign(replUnlockedCryptoService.CryptoService, signedTestRoot, origRootKey)
+	assert.NoError(t, err)
+
+	//
+	// This call to ValidateRoot will succeed since we are usign a valid PEM
```
Comment (NathanMcCauley) [inline keystoremanager/keystoremanager_test.go]: using

File: keystoremanager/keystoremanager_test.go [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+	// encoded certificate, and have no other certificates for this CN
+	//
+	err = keyStoreManager.ValidateRoot(signedTestRoot, gun)
+	assert.NoError(t, err)
+
+	// Finally, validate the the only trusted certificate that exists is the new one
```
Comment (NathanMcCauley) [inline keystoremanager/keystoremanager_test.go]: the the

File: keystoremanager/keystoremanager_test.go [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+	assert.NoError(t, err)
+
+	//
+	// This call to ValidateRoot will succeed since we are usign a valid PEM
+	// encoded certificate, and have no other certificates for this CN
+	//
+	err = keyStoreManager.ValidateRoot(signedTestRoot, gun)
+	assert.Error(t, err, "insuficient signatures on root")
+
+	// Finally, validate the the only trusted certificate that exists is still
```
Comment (NathanMcCauley) [inline keystoremanager/keystoremanager_test.go]: the the

File: keystoremanager/keystoremanager_test.go [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
```
Comment (NathanMcCauley) [inline keystoremanager/keystoremanager_test.go]: the the

**Comment by endophage** [comment]
LGTM

Was thinking on the train home we should check that when the new sig is missing, the new cert doesn't get added to the store by accident, but the existing checks (only 1 cert in store and it's still the original cert) are sufficient to guarantee that :-)

**Comment by NathanMcCauley** [comment]
LGTM after comment updates

**Pull request merged**